include/stencils.hxx

```diff
@@ -35,57 +32,9 @@
 
 /// Defines a set of values in 1D in the neighbourhood of an index
 /// Used for calculating derivatives
-class stencil {
- public:
-  int jx = 0, jy = 0, jz = 0; ///< central location
-  
-  BoutReal mm = BoutNaN, m = BoutNaN, c = BoutNaN, p = BoutNaN, pp = BoutNaN; ///< stencil 2 each side of the centre -- in effect means M?G>2 is invalid
-  
-  /// constructor
-  stencil(){};
-  /// Constructor like pre r115 upwind methods (debugging)
-  stencil(BoutReal fc);
-  /// Constructor like pre r115 derivative methods
-  stencil(BoutReal fc, BoutReal fm, BoutReal fp, BoutReal fmm, BoutReal fpp);
-
-  /// Copy constructor
-  stencil(const stencil &s);
-
-  // operators
-  
-  stencil & operator=(const stencil &rhs);
-  stencil & operator=(BoutReal rhs);
-
-  stencil & operator+=(const stencil &rhs);
-  stencil & operator+=(BoutReal rhs);
-  stencil & operator-=(const stencil &rhs);
-  stencil & operator-=(BoutReal rhs);
-  stencil & operator*=(const stencil &rhs);
-  stencil & operator*=(BoutReal rhs);
-  stencil & operator/=(const stencil &rhs);
-  stencil & operator/=(BoutReal rhs);
-
-  const stencil operator+(const stencil &other) const;
-  const stencil operator+(BoutReal other) const;
-  const stencil operator-(const stencil &other) const;
-  const stencil operator-(BoutReal other) const;
-  const stencil operator*(const stencil &other) const;
-  const stencil operator*(BoutReal other) const;
-  const stencil operator/(const stencil &other) const;
-  const stencil operator/(BoutReal other) const;
-
-  friend const stencil operator+(BoutReal lhs, const stencil &rhs);
-  friend const stencil operator-(BoutReal lhs, const stencil &rhs);
-  friend const stencil operator*(BoutReal lhs, const stencil &rhs);
-  friend const stencil operator/(BoutReal lhs, const stencil &rhs);
-  
-  BoutReal min() const; ///< Minimum value over stencil
-  BoutReal max() const; ///< Maximum value over stencil
-  const stencil abs() const; ///< Stencil with all values replaced by absolute values
+struct stencil {
+  ///< stencil 2 each side of the centre -- in effect means M?G > 2 is not supported
```

Comment:
Very minor quibble, but the doxygen comment should just be `///` to refer to the next line.

Also reading this comment makes me this @JosephThomasParker is on to something with replacing `stencil` with a `vector`. We'd have to do something like `stencil[mesh->xstart]` to get the centre point, so does require some thinking about the interface